File: src/main/java/social/alone/server/event/repository/EventRepository.java
package social.alone.server.event.repository;

import social.alone.server.event.Event;
import social.alone.server.user.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface EventRepository extends JpaRepository<Event, Long>, EventRepositoryCustom {

  Page<Event> findByOwner(User user, Pageable pageable);

  Page<Event> findByUsersContaining(User user, Pageable pageable);
}

File: src/main/kotlin/social/alone/server/event/repository/EventRepository.kt
package social.alone.server.event.repository

import social.alone.server.event.domain.Event
import social.alone.server.user.domain.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor

interface EventRepository :
        JpaRepository<Event, Long>,
        QuerydslPredicateExecutor<Event>,
        EventRepositoryCustom
{

    fun findByOwner(user: User, pageable: Pageable): Page<Event>

    fun findByUsersContaining(user: User, pageable: Pageable): Page<Event>
}
